src/launchpad/insights/common.py

```diff
@@ -0,0 +1,80 @@
+"""Base classes for app bundle insights."""
+
+from __future__ import annotations
+
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Dict, List, Protocol, TypeVar
+
+from launchpad.models.common import DuplicateFilesInsightResult, FileAnalysis, FileInfo
+from launchpad.models.treemap import TreemapResults
+
+from ..models.apple import AppleAppInfo, MachOBinaryAnalysis
+
+T_co = TypeVar("T_co", covariant=True)
+
+
+@dataclass
+class InsightsInput:
```

Comment:
Created a new data class to pass in as an input rather than the weirdness of before with trying to re-use the results model. That model will stay strictly for the final serialization.